xbmc/platform/darwin/tvos/TVOSTopShelf.mm

```diff
@@ -46,119 +46,140 @@
-              [videosArray addObject:@{
-                @"title" : @(title.c_str()),
+              auto itemTitle = getTitleForItem(item);
+              CLog::Log(LOGDEBUG, "[TopShelf] Adding item '{}' in category '{}'",
+                        itemTitle.c_str(), categorKey.UTF8String);
+              [categoryItems addObject:@{
+                @"title" : @(itemTitle.c_str()),
                 @"thumb" : @(fileName.c_str()),
-                @"url" : @(Base64::Encode(videoItem->GetVideoInfoTag()->GetPath()).c_str())
+                @"url" : @(Base64::Encode(item->GetVideoInfoTag()->GetPath()).c_str())
               }];
             }
           }
-          [sharedDefaults setObject:videosArray forKey:videosKey];
-          sharedDictJailbreak[videosKey] = videosArray;
 
-          auto tvTitle = @(g_localizeStrings.Get(titleStringCode).c_str());
-          [sharedDefaults setObject:tvTitle forKey:videosTitleKey];
-          sharedDictJailbreak[videosTitleKey] = tvTitle;
+          // Store category items array in category dict
+          [categoryDict setObject:categoryItems forKey:@"categoryItems"];
+
+          // Store category dict in shared dict
+          if (isSandboxed)
+            [sharedDefaults setObject:categoryDict forKey:categorKey];
+          else
+            sharedJailbreak[categorKey] = categoryDict;
         };
 
-    fillSharedDicts(movies, @"movies", @"moviesTitle", 20386,
-                    [](CFileItemPtr videoItem) {
-                      if (videoItem->HasArt("poster"))
-                      {
-                        return videoItem->GetArt("poster");
-                      }
-                      else
-                        return videoItem->GetArt("thumb");
-                    },
-                    [](CFileItemPtr videoItem) { return videoItem->GetLabel(); });
-
-    CVideoDatabase videoDb;
-    videoDb.Open();
-    fillSharedDicts(tv, @"tv", @"tvTitle", 20387,
-                    [&videoDb](CFileItemPtr videoItem) {
-                      int season = videoItem->GetVideoInfoTag()->m_iIdSeason;
-                      return season > 0 ? videoDb.GetArtForItem(season, MediaTypeSeason, "poster")
-                                        : std::string{};
-                    },
-                    [](CFileItemPtr videoItem) {
-                      return StringUtils::Format(
-                          "%s s%02de%02d", videoItem->GetVideoInfoTag()->m_strShowTitle.c_str(),
-                          videoItem->GetVideoInfoTag()->m_iSeason,
-                          videoItem->GetVideoInfoTag()->m_iEpisode);
-                    });
-    videoDb.Close();
-
-    // remove unused thumbs from cache folder
+
+    // Based on category type, add items in TopShelf shared dict
+    if (category == TVOSTopShelfItemsCategory::MOVIES)
+    {
+      fillSharedDicts(
+          items, @"movies", 20386,
+          [](CFileItemPtr videoItem) {
+            if (videoItem->HasArt("poster"))
+            {
```

Comment:
either drop curly braces or strip down to `return videoItem->GetArt(videoItem->HasArt("poster") ? "poster" : "thumb");`